src/main/java/com/amazonaws/encryptionsdk/CMMHandler.java

```diff
@@ -0,0 +1,97 @@
+package com.amazonaws.encryptionsdk;
+
+import com.amazonaws.encryptionsdk.internal.Utils;
+import com.amazonaws.encryptionsdk.model.DecryptionMaterialsHandler;
+import com.amazonaws.encryptionsdk.model.DecryptionMaterialsRequest;
+import com.amazonaws.encryptionsdk.model.EncryptionMaterialsHandler;
+import com.amazonaws.encryptionsdk.model.EncryptionMaterialsRequest;
+import com.amazonaws.encryptionsdk.model.KeyBlob;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import software.amazon.cryptography.materialproviders.ICryptographicMaterialsManager;
+import software.amazon.cryptography.materialproviders.model.DecryptMaterialsInput;
+import software.amazon.cryptography.materialproviders.model.DecryptMaterialsOutput;
+import software.amazon.cryptography.materialproviders.model.EncryptedDataKey;
+import software.amazon.cryptography.materialproviders.model.GetEncryptionMaterialsInput;
+import software.amazon.cryptography.materialproviders.model.GetEncryptionMaterialsOutput;
+
+/**
+ * Handler to abstract the differences between the original {@link CryptoMaterialsManager} and the
+ * MPL's {@link ICryptographicMaterialsManager}.
+ */
+public class CMMHandler {
+  CryptoMaterialsManager cmm = null;
+  ICryptographicMaterialsManager mplCMM = null;
+
+  public CMMHandler(CryptoMaterialsManager cmm) {
+    Utils.assertNonNull(cmm, "cmm");
+    this.cmm = cmm;
+  }
+
+  public CMMHandler(ICryptographicMaterialsManager mplCMM) {
+    Utils.assertNonNull(mplCMM, "mplCMM");
+    this.mplCMM = mplCMM;
+  }
+
+  public EncryptionMaterialsHandler getMaterialsForEncrypt(EncryptionMaterialsRequest request) {
+    if (cmm != null && mplCMM == null) {
+      return new EncryptionMaterialsHandler(cmm.getMaterialsForEncrypt(request));
+    } else {
+      GetEncryptionMaterialsInput input = getEncryptionMaterialsRequestInput(request);
+      GetEncryptionMaterialsOutput output = mplCMM.GetEncryptionMaterials(input);
+      return new EncryptionMaterialsHandler(output.encryptionMaterials());
+    }
+  }
+
+  private GetEncryptionMaterialsInput getEncryptionMaterialsRequestInput(
+      EncryptionMaterialsRequest request) {
+    return GetEncryptionMaterialsInput.builder()
+        .encryptionContext(request.getContext())
+        .algorithmSuiteId(
+            request.getRequestedAlgorithm() == null
+                ? null
+                : request.getRequestedAlgorithm().getAlgorithmSuiteId())
+        .commitmentPolicy(request.getCommitmentPolicy().getMplCommitmentPolicy())
+        .maxPlaintextLength(request.getPlaintextSize())
+        .build();
+  }
+
+  public DecryptionMaterialsHandler decryptMaterials(
```

Comment:
nit: public methods should have javadoc comments

Reply:
Since these classes are just handlers to abstract the differences between the original class and mpl class. I do not think if we need to add javadoc comments.